snapshot.go

```diff
@@ -5,17 +5,10 @@ import "C"
 
 // Snapshot provides a consistent view of read operations in a DB.
 type Snapshot struct {
-	c   *C.rocksdb_snapshot_t
-	cDb *C.rocksdb_t
+	c *C.rocksdb_snapshot_t
 }
 
 // NewNativeSnapshot creates a Snapshot object.
-func NewNativeSnapshot(c *C.rocksdb_snapshot_t, cDb *C.rocksdb_t) *Snapshot {
-	return &Snapshot{c, cDb}
-}
-
-// Release removes the snapshot from the database's list of snapshots.
-func (s *Snapshot) Release() {
-	C.rocksdb_release_snapshot(s.cDb, s.c)
-	s.c, s.cDb = nil, nil
+func NewNativeSnapshot(c *C.rocksdb_snapshot_t) *Snapshot {
```

Comment:
Jep, it depends whether you prefer to be consistent with the behavior of other code/go structs or with the C API. From another point of view: Carrying dependencies to the underlying rocksdb_transactiondb_t and rocksdb_t around to stick with consistent behavior is inconsistent from the db side. But I'm not sure here. ;)